flink-runtime/src/main/java/org/apache/flink/runtime/resourcemanager/slotmanager/FineGrainedSlotManager.java

```diff
@@ -594,6 +608,30 @@ public int getNumberPendingSlotRequests() {
     // Internal periodic check methods
     // ---------------------------------------------------------------------------------------------
 
+    private void triggerCheckResourceRequirements() {
+        if (resourceRequirementsCheck == null) {
+            resourceRequirementsCheck =
+                    scheduledExecutor.schedule(
+                            () ->
+                                    mainThreadExecutor.execute(
+                                            () -> {
+                                                checkResourceRequirements();
+                                                resourceRequirementsCheck = null;
+                                            }),
+                            requirementsCheckDelay.toMilliseconds(),
+                            TimeUnit.MILLISECONDS);
+        }
+    }
+
+    @VisibleForTesting
+    void manuallyTriggerCheckResourceRequirementsIfNeeded() {
```

Comment:
If we pass in the delay as an argument, we might not need this. Instead, we can pass in a `0` delay.
Additionally, we can leverage `ManuallyTriggeredScheduledExecutor` if needed.